src/elexmodel/models/BaseElectionModel.py

```diff
@@ -53,26 +53,28 @@ def fit_model(self, model, df_X, df_y, tau, weights, normalize_weights):
             LOG.warning("Warning: solution was inaccurate or solver broke. Re-running with normalize_weights=False.")
             model.fit(X, y, tau_value=tau, weights=weights, lambda_=self.lambda_, normalize_weights=False)
 
-    def get_unit_predictions(self, reporting_units, nonreporting_units, estimand):
+    def get_unit_predictions(self, reporting_units, nonreporting_units, estimand, **kwargs):
         """
         Produces unit level predictions. Fits quantile regression to reporting data, applies
         it to nonreporting data. The features are specified in model_settings.
         """
-        # compute the means of both reporting_units and nonreporting_units for centering (part of featurizing)
-        # we want them both, since together they are the subunit population
-        self.featurizer.compute_means_for_centering(reporting_units, nonreporting_units)
-        # reporting_units_features and nonreporting_units_features should have the same
-        # features. Specifically also the same fixed effect columns.
-        reporting_units_features = self.featurizer.featurize_fitting_data(
-            reporting_units, add_intercept=self.add_intercept
+        n_train = reporting_units.shape[0]
+        n_test = nonreporting_units.shape[0]
+        all_units = pd.concat([reporting_units, nonreporting_units], axis=0)
+
+        featurizer = Featurizer(self.features, self.fixed_effects)
+        x_all = featurizer.prepare_data(
```

Comment:
But then you're leaking data from the training data into the held-out/testing data 😞

Reply:
That doesn't matter though, because that information is actually always known about the data

Reply:
Hmmmmmmmmm...ok, I think I'm ok with this, if:

1. when we do evaluations, we report metrics with an asterisk, or we do evaluations this way and not this way, so we have two MAPEs, two MAEs, etc. per whatever we're evaluating (e.g. county _X_ at time _t_);
2. by this logic, it should be ok to use the entire data set to create fixed effect dummies; see below 😄

Reply:
Sure, but I don't think I understand why it's necessary in this case. We're always in a situation where we know things about the held out data (the demographic features), so we're never doing anything with that data that we're not allowed too. Leaking information is only a problem if you're testing data leaks into our training data, but that isn't the case here. Our held-out data is actually the data that we are applying the model to in production.

Reply:
Hmmmmmmm...ok, that makes sense and it's interesting because I've never thought of it that way before.  I can only think of models I've worked on in the past where live prediction would be performed far enough into the future, to standardize with the testing data included in the standardization would give a false sense of generalizability 🤔 😄